### PR TITLE
Accept multiple X509 certificates and parse them correctly

### LINKED ISSUE #7: Failure to parse IdP certificate
Following the commit from @RavuAlHemio to add signature verification (commit 816c6ab), I had an issue with the `idp_signing_certs()`.

I got `FailedToParseCert` when testing with http://samltest.id.

I found that in there metadata the x509 certificate data was wrapped making the loading fail.
As a workaround I need to do:

```Rust
    samael::metadata::de::from_str(&raw_metadata)
        .map(|mut ed: EntityDescriptor| {
            if let Some(descriptors) = &mut ed.idp_sso_descriptors {
                for descriptor in descriptors.iter_mut() {
                    for key in descriptor.key_descriptors.iter_mut() {
                        if let Some(data) = &mut key.key_info.x509_data {
                            if let Some(certificate) = &mut data.certificate {
                                *certificate = certificate.replace("\n", "");
                            }
                        }
                    }
                }
            }
            ed
        }
```

I'm wondering if the fix might not simply be to replace `base64::decode(cert.as_bytes())` by `decode_config(cert.as_bytes(), base64::STANDARD.decode_allow_trailing_bits(true))`. 

In any case thanks for the job you are doing. Weirdly this is the only SAML crate I could find.

### PR DESCRIPTION
**Pull request by samueltardieu**
They will be parsed correctly even if they are spread on several lines.

Fix #7 and #11

### CODE REVIEW

File: src/idp/verified_request.rs [hunk 1]
```diff
@@ -16,31 +16,39 @@ impl<'a> UnverifiedAuthnRequest<'a> {
...
+    pub fn get_certs_der(&self) -> Result<Vec<Vec<u8>>, Error> {
+        let x509_certs = self
             .request
             .signature
             .as_ref()
             .ok_or(Error::NoSignature)?
             .key_info
             .as_ref()
-            .map(|ki| ki.iter().next())
+            .map(|ki| ki.iter().next()) // TODO: why only the first key?
```
Comment (njaremko) [inline src/idp/verified_request.rs:27]: Only the first key was just lazy writing initially

**Review by njaremko — COMMENTED** [review]
Thanks for this, I really appreciate it. I didn't see the notification and it got buried by PRs from work. 

I've left a few comments, but looks good overall 👍

**Comment by samueltardieu** [comment]
> Thanks for this, I really appreciate it. I didn't see the notification and it got buried by PRs from work.

Hi. I left interest in this patch as I have made other heavy modifications in my fork since (including not depending on yaserde whose repository is not even in a compilable state and doesn't contain any test). I'll close this PR, feel free to take bits if needed.

**Comment by Mikescops** [comment]
Hey @njaremko, thanks for your work!
Would you move forward with this issue? those `\n` are pretty blocking the whole interest of the project :grin:

**Comment by njaremko** [comment]
Hi @Mikescops, I'm going to pick this back up 👍

**Comment by Mikescops** [comment]
Thanks 😃

**Pull request merged**